docker/terra-node/Dockerfile

```diff
@@ -1,16 +1,26 @@
-# use node as base image
-FROM node:8.9.2
+FROM node:8
 
-# Prepare Directory for Source Code
-ENV APP_HOME /app
-RUN mkdir $APP_HOME
-WORKDIR $APP_HOME
+EXPOSE 8080
 
-# Mount Source into Container
-ONBUILD ADD . /app
+COPY ./entrypoint /opt/entrypoint
 
-# expose the port
-EXPOSE 8080
+RUN npm config set unsafe-perm true -g
+
+#install zsh / oh-my-zsh
+RUN apt-get update \
+  && apt-get install -y zsh \
+  && apt-get clean \
+  && wget https://github.com/robbyrussell/oh-my-zsh/raw/master/tools/install.sh -O - | zsh || true
+
+# Do not set a default here or the building docker file will not be able to override the variable
+ONBUILD ARG project_name
+ONBUILD ARG npm_install
+
+# Copy all the things
+ONBUILD COPY . /opt/${project_name:-module}
```

Comment:
should we change opt -> app for naming consistency with other docker files consumers use

Reply:
by default it'll be opt -> module. In some cases (terra-dev-site) changing the directory name will cause tests to fail. So the project_name env was added to allow docker containers to set then name as they see fit and minimize refactoring to consume this change.

Reply:
I was referencing the `/opt/` part of this, not the `{project-name:-module}` naming. Most dockerfiles of our consumers have `app` as the home directory, which would end up with `/app/{project-name:-module}`

Reply:
I think thats a cerner convention. We don't have an app, we have a module.

Reply:
True, this isn't an app. I does seem like a cerner convention to use `app`